framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4240,6 +4291,202 @@ private void makeGroundTargetType(
         addDefaultAnnotations(functionalType);
     }
 
+    /**
+     * Return true if {@code type} should be captured.
+     *
+     * <ul>
+     *   <li>{@code type} and {@code typeMirror} must both be declared types.
+     *   <li>{@code type} must not be a raw type, must not have an uninferred type argument, and
+     *       must have a wildcard as a type argument.
+     *   <li>{@code typeMirror} must have a captured type as a type argument.
+     * </ul>
+     *
+     * @param type annotated type to capture
+     * @param typeMirror underlying Java type
+     * @return true if {@code type} should be captured
+     */
+    private boolean shouldCapture(AnnotatedTypeMirror type, TypeMirror typeMirror) {
+        if (type.getKind() != TypeKind.DECLARED && typeMirror.getKind() != TypeKind.DECLARED) {
```

Comment:
This test looks wrong to me.  I think the `&&` should be `||`.
However, I would rewrite it to correspond closely to the Javadoc comment `@code type} and {@code typeMirror} must both be declared types.`:

```java
        if (!(type.getKind() == TypeKind.DECLARED && typeMirror.getKind() == TypeKind.DECLARED)) {
            return false;
	}
```

I think a closer correspondence between the Javadoc and the code makes the code easier to understand.